### PR TITLE
Update HASH welcome message

### PR DESCRIPTION
**Pull request by teenoh**
## 🌟 What is the purpose of this PR?

This PR updates the welcome copy of the HASH app

## 🔗 Related links

- [Update HASH welcome message](https://app.asana.com/0/1202250835094696/1202381444789113) _(internal)_

## 🔍 What does this change?

- Updates the welcome message users see when logged into the app

## 📜 Does this require a change to the docs?

- No

## ❓ How to test this?

1.  Checkout the branch
1.  Login. The welcome page should match attached screenshot

## 📹 Demo

Screenshot 2022-06-06 at 10 22 02

### CODE REVIEW

File: packages/hash/frontend/src/pages/[account-slug]/index.page.tsx [hunk 1]
```diff
@@ -46,21 +48,45 @@ const Page: NextPageWithLayout = () => {
...
-          Hi, {user.properties.preferredName}!
+      <Typography mb={3} variant="h2">
+        Welcome to HASH
+      </Typography>
+      <Box sx={{ maxWidth: "75ch" }}>
+        <Typography mb={3}>
+          HASH is an open-source, data-centric, all-in-one workspace built atop
+          the open <Link href="https://blockprotocol.org">Block Protocol</Link>,
+          allowing users to easily add new block types and functionality to
+          their workspaces.
```
Comment (nonparibus) [inline packages/hash/frontend/src/pages/[account-slug]/index.page.tsx:61]:
```suggestion
          the open <Link href="https://blockprotocol.org">Block Protocol</Link>.
```

Comment (teenoh) [reply packages/hash/frontend/src/pages/[account-slug]/index.page.tsx]: Updated in [c40fbef](https://github.com/hashintel/hash/pull/610/commits/c40fbef90c42ce84c851ca57ec9676a172480d84)

File: packages/hash/frontend/src/shared/ui/link.tsx [hunk 2]
```diff
@@ -10,10 +10,10 @@ import { styled } from "@mui/material/styles";
 import { Button } from "./button";
 import { FRONTEND_URL } from "../../lib/config";
 
+// @todo: update the regex to check against the domain of the hosted version of HASH
 export const isHrefExternal = (href: string | UrlObject) =>
   typeof href === "string" &&
-  (href === "/discord" ||
-    !/^(mailto:|#|\/|https:\/\/blockprotocol\.org)/.test(href)) &&
+  (href === "/discord" || !/^(mailto:|#|\/)/.test(href)) &&
```
Comment (kachkaev) [inline packages/hash/frontend/src/shared/ui/link.tsx:16]: If I understand correctly, BP links are external now (i.e. open in a new tab). That makes sense, but I don't quite remember the reason for making them ‘internal’ in the first place.

**Review by kachkaev — APPROVED** [review]
The new copy makes sense! I left a couple of comments, but they don't block the merge

File: packages/hash/frontend/src/pages/[account-slug]/index.page.tsx [hunk 3]
```diff
@@ -46,21 +48,43 @@ const Page: NextPageWithLayout = () => {
...
+            />{" "}
+            HASH is not ready for production use.
+          </strong>{" "}
+          It is not secure or optimized and is missing key features. Please
+          visit the{" "}
+          <Link href="https://github.com/hashintel/hash/tree/main/packages/hash">
+            GitHub repository
+          </Link>{" "}
+          for the latest updates, or learn about the long-term{" "}
+          <Link href="https://hash.ai">here.</Link>
```
Comment (kachkaev) [inline packages/hash/frontend/src/pages/[account-slug]/index.page.tsx:78]: Might be worth aligning `<a>link</a>.` and `<a>link.</a>`

- `<Link href="https://blockprotocol.org">Block Protocol</Link>.`
- `<Link href="https://hash.ai">here.</Link>`
- `<Link href="https://github.com/hashintel/hash/tree/main/packages/hash#integration-with-the-block-protocol">read the documentation to get started.</Link>`

I’m in favor of `<a>link</a>.` but happy with any option as long as it’s consistent.

**Review by kachkaev — CHANGES_REQUESTED** [review]
Let’s go for `<a>link</a>.` everywhere 🙂

The branch is out of date, so needs `git push` anyway 👀

File: packages/hash/frontend/src/shared/ui/link.tsx (hunk 2 above)
Comment (teenoh) [reply packages/hash/frontend/src/shared/ui/link.tsx:16]: When building out the base components, we copied over some base code from the BP site and forgot to update this to track blockprotocol.org as an external link.

File: packages/hash/frontend/src/shared/ui/link.tsx [hunk 4]
```diff
@@ -10,10 +10,10 @@ import { styled } from "@mui/material/styles";
 import { Button } from "./button";
 import { FRONTEND_URL } from "../../lib/config";
 
+// @todo: update the regex to check against the domain of the hosted version of HASH
```
[the remark this answers is not in the document]
Comment (teenoh) [reply packages/hash/frontend/src/shared/ui/link.tsx:13]: This is meant to update the regex to make links of the hosted HASH domain internal, just like we do for `https://blockprotocol.org/*` in the BP site. I wasn't sure of the final domain we'd use for that (I know we use alpha.hash.ai at the moment) and decided to add the todo

**Review by kachkaev — DISMISSED** [review]

**Review by kachkaev — APPROVED** [review]

**Pull request merged**